cmd/s3-gw/app_metrics.go

```diff
@@ -63,18 +72,44 @@ func newStateMetrics() *stateMetrics {
 	}
 }
 
-func (m stateMetrics) register() {
-	prometheus.MustRegister(m.healthCheck)
-}
-
-func (m stateMetrics) unregister() {
+func (m *stateMetrics) unregister() {
 	prometheus.Unregister(m.healthCheck)
 }
 
-func (m stateMetrics) SetHealth(status healthStatus) {
+func (m *stateMetrics) SetHealth(status healthStatus) {
 	m.healthCheck.Set(float64(status))
 }
 
+func (m *stateMetrics) updateHealthStatus() {
+	ctx, cancel := context.WithTimeout(context.Background(), checkNeoFSConnectionTimeOut)
+	defer cancel()
+
+	cl, err := m.p.RawClient()
+	// Only "no healthy client" error is possible.
+	if err != nil {
+		m.SetHealth(healthStatusUnhealthy)
```

Comment:
i'd dedicate separate metric for this. Some kind of conn probe, unlike existing one reflecting app stages. Moreover, it relies on single connection. How this gonna behave? If we pool 9 healthy conns and 1 unhealthy one, can we unhealthy here? Cuz if so, the metric will poorly reflect the likelihood of OK request